client/tokenized-express-checkout/index.js

```diff
@@ -440,13 +446,6 @@ jQuery( ( $ ) => {
 				wcpayExpressCheckoutParams.product = undefined;
 			}
 
-			if ( getExpressCheckoutData( 'button_context' ) === 'product' ) {
```

Comment:
Moved to the `click` handler, so that the `click` handler will always reset the state of the cart and initialize a new cart session in case of error.